### ISSUE TITLE
Upgrade Oboe to latest commit to fix CodeQL errors

### ISSUE THREAD

**Issue by ann0see** [jamulussoftware/jamulus#2545]
**What is the current behaviour and why should it be changed?**
Currently CodeQL fails due to Oboe issues. Since https://github.com/google/oboe/pull/1507 got merged they should be fixed

**Describe possible approaches**
* Wait for official release and live with warnings

*  backport changes

* pin the current master commit  

**Has this feature been discussed and generally agreed?**
No

**Comment by hoffie** [comment]
Hrm, this would be very good to have.

Current master has a large amount of changes compared to the last released version (which we currently use):

```
~/Projects/jamulus/libs/oboe$ git diff --stat HEAD..origin/main
[...]
161 files changed, 3262 insertions(+), 914 deletions(-)
```

Backporting would imply including the source instead of using a submodule, I guess? Not nice either :(

**Comment by pljones** [comment]
Can you list the merges, rather than the diff?  That might be a quick start.

**Comment by hoffie** [comment]
No merges there, therefore, a one-line log:
<details>
<summary>~/Projects/jamulus/libs/oboe$ git log --format=one HEAD..origin/main</summary>

```
6adcc5d647277f0969def610ceb5f6a8b703dc81 AAudio: add locks around reading and writing (#1497)
2d4797ad433aa220499d7ea55f4d15a667decc27 Add static code analysis on commit and fix CodeQL errors (#1507)
cdbd175dc4fd435bacc77f678a4f1dd6d7178a91 hi hat exclusive (#1503)
126c0aace2d34e72b6e002fada3ead7611269034 RhythmGame: Add some instructions (#1505)
516c9d1bebd25ee611774e9d0328445ba836eada Use Java 11 in CI
aa40d86527a815570c0f9933a35b288f2abf7038 Update gradle in samples
1a7dce497ded081ed127059ae46455aa105612b6 Recommend not calling getTimestamp() from callback (#1499)
d50ea028ee4b3cce99ef87aadbc3ba6c78741e45 Fix some compile errors in the full example (#1495)
ad9052e6c9aa73cba9fe58bcde4452c86848533b Bump minSdkVersion to 21 and targetSdkVersion to 31 (#1491)
56bfc8b301a1501b68c6e12c411985887ca655cd OboeTester: show frames in hex (#1488)
a0731d6041561da9a15efdfd4ed49718b57bf5b4 use shared_ptr in FullGuide
7b3d34ab18b1051950693daee7e45836e18b32a7 (HelloOboe) Add Bluetooth SCO sample (#1471)
b161fd170e1f92a1fb13402850159a6180d0b54f link to tech note in Wiki
7d25f15f32c0e2f495d51262e0f4ab8f53fde63d remove glitches test note, moved to Wiki
8638040796c38a3a1aab05b57f02afcab58ff217 Add link to glitches.md
92a2292e89c2b3eeb6f4223fbf140dd20b6bff17 Add Tech Note on glitches
b920c8161b7aa762a5481a8607c088b1b824e95d sync flowgraph with internal repo (#1464)
23c7bf6213bdd238af814f479dadc160765bb835 add cast to setSessionId doc
9a9487a82a771ea8a7705861129a65df3da147c1 Fix 1407 sync flowgraph (#1416)

adfa2665934d08cc84e079bd75f28a1d8cd695b7 Bump Oboe version to 1.6.2 (#1437)    <--- There's no tag though. --@hoffie
02b53b946ec747c99d20daeb1e56f94da550c57f (LiveEffect) Disable setting playback or recording devices with OpenSL ES (#1424)
37aafe6f4a9ce3dc81b7b15726ccf80e395cf3df (drumthumper) remove redundant semicolon in kotlin (#1432)
207508f6f0f8b8d572bce91db7f957e02ec29b49 Add more info for setFramesPerDataCallback (#1426)
543e8e9747930bd44fdf13ba4d63006b737f50ba (OboeTester) Add sine amplitude to glitch test (#1422)
1f02fcf87ecffeb3f2ba66cd5c7eb99d3b0ad304 use calculateLatencyMillis in hello oboe (#1423)
16c468dd05966bd2e686f2086181f90b6eefde8c add link to privacy policy
a887d522ad40e90e16678c8e7ac29e1746354f27 add privacy policy for Oboe
208e1a714ad9df13f07e05e009f0e5e44d8cd7eb correct spelling
4b5f0156e479f1ac1a60db97ced46355d6e1ee76 Mention limitations of AudioStreamErrorCallback in documentation
23ee4b6fa2fe14327d329a7ad1e5469f4819ec22 Move FifoBuffer header in public directory (#1410)
28de9887e7cb2e808792b32f9bf6f0d267ce6d46 (OboeTester) Add Content Type Menu (#1408)
6c9513a4db05dc65b06905d02555e5a7e11a9d4f (OboeTester) Add Test Plug Latency Activity (#1403)
83f79e93e32dee9e2b5d3ced07dc94c82d4bc182 (OboeTester) Add time between callback statistics (#1394)
5ca07db3475e025608ed9b260b35dc928fe8b873 Quirk: disable MMAP on SDM8150 (#1381)
707069149c9f7b6a22017ae5cd266598e3ba3563 OboeTester: create automatic data paths tests (#1390)
dc99c18398c2a63f647cb6610327be3ce40a4737 OboeTester: Add time when tap to tone is pressed to logs (#1388)
4c728bcb7983eab63056945d6cd67abd53ff5691 fix sawtooth waves with test output (#1392)
07193e73af5f1ca0672060592ca320c7c082c3d0 fix permission issues for test glitches (#1386)
88f2d79047877a54b2933f15b665d916f02bfe5d Delete NOTICE file
8ed0c1ad0ca15a6d7a41f5c4078c88e8444f04c7 Change headings for Disconnects note
0eac17715f3166965ce7dab70efd7f0f677fff10 Update Disconnects note
7a64657565fca3fab7078b3156bed198118e1f09 OboeTester: bump version to 2.2.2
71fa9343930189c7f6cc7583014389ed70e34dbb OboeTester: suggest raising volume (#1378)
88237d982c032210ce6babd28943eb9e610db7cb OboeTester: prevent taps when stopped (#1377)
fbfebac3883b79e1cec617ce91db82d062b87199 OboeTester: improve data paths test (#1376)
3c17ec5e937c945e92f798f3649abbca7ebe3d04 OboeTester: improve VU meter display (#1374)
24d170167d12ea40b99e3ede1f40c7e7358be7dc Soundboard app (#1354)
18bb82d3ae54b7a3ebbc3236fbd0e498444d2950 OboeTester: fix closest test search (#1373)
7933a7ed1f4130ca00581159ff819f553afd8105 Use an array of callback buffer for OpenSL ES. (#1245)
ee267abce64ed5c28a2002f6a09050eee4533686 OboeTester: improve readability of DATA PATHS report (#1369)
5dafec69f32e042454d5f41a9d99379d2b2dd45d Add link to Buffer Terminology on Wiki
0211a81968c54c3633b63f0c946f871a44ef4430 OboeTester: refactor RECORD_AUDIO permissions (#1366)
d683731f3c607f25f6d8d29a613dd36f89b08221 remove duplicated header from test files (#1362)
c3218750096a9fd038533343fb7f116920aa3d92 OboeTester: fix MIDI and crash (#1361)
d4826385aa13721f06c56ae94df14d008daf494b Update package name header (#1355)
```
</details>

**Comment by ann0see** [comment]
According to this comment https://github.com/google/oboe/issues/1506#issuecomment-1078165010

the master branch of oboe is clean
